### ISSUE TITLE
No E-Mail notification for submissions marked as spam

### ISSUE THREAD

**Issue by fabhoffmann** [verbb/formie#1599]
### Describe the bug

When a user submits a form and the submissions was marked as spam, no notificaion is sent. We also enabled **Send Email Notifications** in the settings in the spam section. In the Queue-Manager we get the message that no entry could be found under the specified Id. But the associated submission exists.

![error](https://github.com/verbb/formie/assets/148063251/2c4d792f-3fd8-4383-a7ad-6ba2a3a9410f)

We found out that we don't get a submission object back in SendNotification.php if we take the id of an entry marked as spam

```
$submission = Formie::$plugin->getSubmissions()->getSubmissionById($this->submissionId);
```

The reason for this could be in the SubmissionQuery.php:

```
if ($this->isSpam !== null) {
       $this->subQuery->andWhere(Db::parseParam('formie_submissions.isSpam', $this->isSpam));
}
```

In our case isSpam is false although the submission was marked as spam. Accordingly, the condition is set and we do not get our submission back. Maybe the default value for isSpam is no longer correct in Submission.php?

### Steps to reproduce

1. Submit a form
2. Submission is marked as spam
3. Queue tries to send notification

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

4.4.15

### Plugin version

2.0.36

### Multi-site?

Yes

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.38"`.

**Comment by engram-design** [comment]
Fixed in [2.0.39](https://github.com/verbb/formie/releases/tag/2.0.39)
